ui/raidboss/popup-text.ts

```diff
@@ -1077,7 +1077,7 @@ export class PopupText {
 
   _onTriggerInternalCondition(triggerHelper: TriggerHelper): boolean {
     const condition = triggerHelper.triggerOptions.Condition ?? triggerHelper.trigger.condition;
-    if (condition) {
+    if (condition !== undefined && condition !== false) {
```

Comment:
Yeah, I just tested this against main, that's a fun unintentionally found bug 🙃

Reply:
This is the rewrite I'm thinking makes the most sense for this:
```ts
  _onTriggerInternalCondition(triggerHelper: TriggerHelper): boolean {
    const condition = triggerHelper.triggerOptions.Condition ?? triggerHelper.trigger.condition;
    if (condition === undefined || condition === true)
      return true;
    else if (condition === false)
      return false;

    const conditionFuncReturn = condition(this.data, triggerHelper.matches, triggerHelper.output);
    if (conditionFuncReturn === true)
      return true;
    return false;
  }
```

The question is, since `conditionFuncReturn` is of type `boolean | undefined`, what do we do with `undefined`, e.g. what do we do if the condition function returns undefined? Right now, `undefined` is treated as `false`, e.g. abort the trigger because condition failed. The rewritten function above treats it as the same, but maybe a comment to indicate that it's intentional behavior is warranted?

Reply:
Yeah, I think falsey values can be treated as `false` here with a comment.